src/ResourceManager/Sql/Commands.Sql/Common/AzureEndpointsCommunicator.cs

```diff
@@ -231,5 +232,140 @@ private ResourceManagementClient GetCurrentResourcesClient(IAzureContext context
             }
             return ResourcesClient;
         }
+
+        /// <summary>
+        /// Retrieves storage keys.
+        /// </summary>
+        /// <param name="storageAccountSubscriptionId">Storage account subscription id</param>
+        /// <param name="storageAccountName">Storage account name</param>
+        /// <returns>Dictionary containing storage keys</returns>
+        internal Dictionary<StorageKeyKind, string> RetrieveStorageKeys(Guid storageAccountSubscriptionId, string storageAccountName)
+        {
+            // Retrieve the id of the storage account.
+            //
+            string storageAccountId = RetrieveStorageAccountIdAsync(storageAccountSubscriptionId, storageAccountName).GetAwaiter().GetResult();
+
+            // Extract storage account keys. 
+            //
+            return RetrieveStorageKeysAsync(storageAccountId).GetAwaiter().GetResult();
+        }
+
+        /// <summary>
+        /// Retrieves storage account keys.
+        /// </summary>
+        /// <param name="storageAccountId">Storage account id</param>
+        /// <returns>Dictionary containing storage keys</returns>
+        private async Task<Dictionary<StorageKeyKind, string>> RetrieveStorageKeysAsync(string storageAccountId)
+        {
+            bool isClassicStorage = storageAccountId.Contains("Microsoft.ClassicStorage/storageAccounts");
+
+            // Build a URI for calling corresponding REST-API
+            //
+            StringBuilder uriBuilder = new StringBuilder(Context.Environment.GetEndpointAsUri(AzureEnvironment.Endpoint.ResourceManager).ToString());
+            uriBuilder.AppendFormat("{0}/listKeys?api-version={1}",
+                storageAccountId,
+                isClassicStorage ? "2016-11-01" : "2017-06-01");
+
+            // Define an exception to be thrown on failure.
+            //
+            Exception exception = new Exception(string.Format(Properties.Resources.RetrievingStorageAccountKeysFailed, storageAccountId));
+
+            // Call the URI and get storage account keys.
+            //
+            JToken storageAccountKeysResponse = await SendAsync(uriBuilder.ToString(), HttpMethod.Post, exception);
+
+            // Extract keys out of response.
+            //
+            Dictionary<StorageKeyKind, string> storageAccountKeys = new Dictionary<StorageKeyKind, string>();
+            string primaryKey = null;
+            string secondaryKey = null;
+            if (isClassicStorage)
+            {
+                primaryKey = (string)storageAccountKeysResponse["primaryKey"];
+                secondaryKey = (string)storageAccountKeysResponse["secondaryKey"];
```

Comment:
Nitpick : These string literals should be constants instead.

Reply:
@praries880 Done.